include/package-ipkg.mk

```diff
@@ -218,7 +219,14 @@ $(_endef)
 	$(if $(PROVIDES),@for pkg in $(filter-out $(1),$(PROVIDES)); do cp $(PKG_INFO_DIR)/$(1).provides $(PKG_INFO_DIR)/$$$$pkg.provides; done)
 	$(CheckDependencies)
 
+    ifneq ($$(CONFIG_EXTRACT_DEBUGINFO),)
+	[ ! -d $$(DBGINFO_$(1)) ] || rm -rf $$(DBGINFO_$(1))
+	mkdir $$(DBGINFO_$(1))
+	export DEBUGINFO_DIR=$$(DBGINFO_$(1)); \
+		$(RSTRIP) $$(IDIR_$(1))
+    else
 	$(RSTRIP) $$(IDIR_$(1))
+    endif
```

Comment:
```suggestion
    ifneq ($$(CONFIG_EXTRACT_DEBUGINFO),)
	[ ! -d $$(DBGINFO_$(1)) ] || rm -rf $$(DBGINFO_$(1))
	mkdir $$(DBGINFO_$(1))
	export DEBUGINFO_DIR=$$(DBGINFO_$(1));
    endif
	$(RSTRIP) $$(IDIR_$(1))
```

Isn't this simpler?

Reply:
The export on (your) line 225 is ineffective because 227 is run in a new shell.
I agree, though, that it would be nice to have only one line invoking `RSTRIP`.
Make variables could probably be used to carry it over (in the 'inner' make scope) but I don't think I'm knowledgeable enough to make it work in a reasonable amount of time.